### ISSUE TITLE
wxHTMLDataObject::GetDataHere implementation causes memory corruption.

### ISSUE THREAD

**Issue by mcorino** [wxWidgets/wxWidgets#23660]
### Description
The implementation of `bool wxHTMLDataObject::GetDataHere(void *buf) const` in 'src/common/dobjcmn.cpp' at line 450 causes memory corruption by using `strcpy()` (at line 505) instead of `memcpy()` as used by wxTextDataObject.

The buffers allocated for textual data objects like wxTextDataObject and wxHTMLDataObject do *not* include space for closing null characters so the use of `strcpy()` causes overwrites of unallocated memory.
Replacing the code `strcpy(buffer, html);` with `memcpy(buffer, html, strlen(html));` solves the issue.

Adding this test

```c++
TEST_CASE("GUI::HTMLDataObject", "[guifuncs][clipboard]")
{
    const wxString text("<h1>Hello clipboard!</h1>");

    wxHTMLDataObject* const dobj = new wxHTMLDataObject(text);
    CHECK( dobj->GetHTML() == text );

    wxClipboardLocker lockClip;
    CHECK( wxTheClipboard->SetData(dobj) );
    wxTheClipboard->Flush();

    wxHTMLDataObject dobj2;
    REQUIRE( wxTheClipboard->GetData(dobj2) );
    CHECK( dobj2.GetHTML() == text );
}
```

to 'tests/misc/guifuncs.cpp' will demonstrate the problem.

With the current code the test will fail like this:

```
Filters: GUI::HTMLDataObject
free(): invalid pointer
Killed
```

With the proposed change the test will succeed.

### Platform and version information

- wxWidgets version  you use: master
- wxWidgets port  you use: wxGTK
- OS  and its version: OpenSuSE 15.4

+ GTK version:  3.24.34
+ Which GDK  backend is used: X11
+ Current theme:  NA

**Comment by vadz** [comment]
Thanks, this is indeed clearly a buffer overrun and ASAN detects it fine, so I'll add this test and the fix.

Please don't hesitate to create PRs with your fixes in the future!

**Comment by vadz** [comment]
Sorry for the confusion but, just for the record, this issue was actually fixed in 1cf83980a2 (Refactor wxHTMLDataObject to keep wxMSW-specific parts together, 2023-06-21) due to sloppy refactoring/rebasing of the commits.
